### PR TITLE
Elva and KiYong Shopping List (Redux)

### PR DESCRIPTION
**Pull request by kiyonggg**
Done:

- Layout of page
- Add items into list (using redux)
- Delete items from list (using redux)
- Change quantity (using redux)
- Validation for adding items (only cannot add if name not included and quantity is less than 1)
Screenshot 2021-10-23 at 11 16 41 PM

### CODE REVIEW

**Review by xinyee20 — COMMENTED** [review]
Package-lock.json should not be present as we are using yarn as package manager, please delete the folder. Something doesn't feel right when you see the number of lines change is 30k ++😳

File: src/pages/ShoppingCart4/index.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+import AddItemModal from '../../components/AddItemModal'
+
+export interface ShoppingListItem {
+  key: number
+  shoppingListItemName: string
+  itemCount: number
+}
+
+const ShoppingCart4 = () => {
+  const [shoppingList, setShoppingList] = useState([] as ShoppingListItem[])
```
Comment (xinyee20) [inline src/pages/ShoppingCart4/index.tsx]: can be written as useState<ShoppingListItem[]>([]) where the type is written in the <>

File: src/App.tsx [hunk 2]
```diff
@@ -23,6 +25,12 @@ const Groups = React.lazy(() => import(/* webpackChunckName: "Groups" */ './page
 const ShoppingListsMainPage = React.lazy(
   () => import(/* webpackChunckName: "ShoppingListsMainPage" */ './pages/ShoppingListsMain'),
 )
+const ShoppingListPair4 = React.lazy(
+  () => import(/* webpackChunckName: "ShoppingListsMainPage" */ './pages/ShoppingCart4'),
```
Comment (xinyee20) [inline src/App.tsx]: update the webpackChunckNames

File: src/App.tsx [hunk 3]
```diff
@@ -45,6 +53,23 @@ function App() {
           <ShoppingListsMainPage />
         </StyledMain>
       </Route>
+      <Route path={`${PATHS.SHOPPING_PAGE_PAIR_FOUR}`} exact>
```
Comment (xinyee20) [inline src/App.tsx]: for those without additional text such as this one, can use path={PATHS.SHOPPING_PAGE_PAIR_FOUR} without the `${}`

File: src/store/shoppingCart4Redux/actions.ts [hunk 4]
```diff
@@ -0,0 +1,59 @@
+import { Dispatch, GetState } from '../types'
+import { ActionTypes, SHOPPING_LIST_ACTIONS, ShoppingListItem } from './types'
+
+//set my 'exampleList' state with my argument 'newList'
```
Comment (xinyee20) [inline src/store/shoppingCart4Redux/actions.ts]: comments can be removed:)

**Review by xinyee20 — CHANGES_REQUESTED** [review]

File: src/pages/ShoppingCart4/index.tsx (hunk 1 above)
Comment (kiyonggg) [reply src/pages/ShoppingCart4/index.tsx]: resolved in commit 6052692

File: src/App.tsx (hunk 2 above)
Comment (kiyonggg) [reply src/App.tsx]: resolved in commit 6052692

File: src/App.tsx (hunk 3 above)
Comment (kiyonggg) [reply src/App.tsx]: resolved in commit 6052692

File: src/store/shoppingCart4Redux/actions.ts (hunk 4 above)
Comment (kiyonggg) [reply src/store/shoppingCart4Redux/actions.ts]: resolved in commit 6052692

File: src/components/AddItemModal.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+  const validateDataError = () => {
+    if (shoppingListItemName.length === 0) {
+      setItemNameError('Please input an item name'!)
+      return true
+    } else {
+      setItemNameError('')
+    }
+    if (itemCount <= 0) {
+      setItemCountError('Please input a valid quantity!')
```
Comment (xinyee20) [inline src/components/AddItemModal.tsx]: Maybe 'Please input a quantity greater than 0!' will be a better error message? Something to indicate what is valid

Comment (kiyonggg) [reply src/components/AddItemModal.tsx]: resolved in commit ab913af

Comment (kiyonggg) [reply src/components/AddItemModal.tsx]: resolved in commit ed6e82a

**Review by xinyee20 — APPROVED** [review]
Lgtm!  Make sure checks pass and conflicts are resolved before merging

**Comment by hui444** [comment]
> Lgtm!

pls make sure checks pass!!!!!!!!

**Review by xinyee20 — APPROVED** [review]
lgtm! Resolve conflicts before merging

**Pull request merged**